onnxruntime/core/session/abi_session_options_impl.h

```diff
@@ -6,6 +6,8 @@
 #include <string>
 #include <vector>
 #include <atomic>
+#include "core/common/status.h"
+#include "core/common/path_string.h"
```

Comment:
I am wondering where it gets included since it is an ABI.
Perhaps, forward declarations for Status and PathString would be more appropriate.

Reply:
We discussed in person and decided to leave "path_string.h" as is.

Reply:
We should not rely on indirect includes. This makes things brittle and introduces breakages when the file that includes it decides to remove the include, only to find that other things depend on it. Every header should be self contained.

Reply:
Agreed. This is the reason why this file now includes "status.h" explicitly.